nixos/modules/services/web-apps/siyuan.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
```

Comment:
Please avoid adding new with libs over entire files.

Reply:
As far as I found, this is a widely used pattern in modules... Several examples in the documentation (https://nixos.org/manual/nixos/stable/index.html#sec-writing-modules) do this.

Could you at least should what you think should be done instead?
Is this written somewhere in a reference or documentation, or is it custom nitpicking?

Reply:
See https://github.com/NixOS/nixpkgs/issues/208242

Top-level `with lib;` was widely used in the past, but is in the process of being phased out of usage. (Though, as you may have seen, not all occurrences have been replaced yet)

An alternative would be to use `lib.` before everything, e.g: `lib.mkOption` instead of `mkOption`.